pkg/rc/replication_controller.go

```diff
@@ -952,13 +953,30 @@ func (rc *replicationController) scheduleNewNodeForTransfer(rcFields fields.RC,
 func (rc *replicationController) doBackgroundNodeTransfer(rcFields fields.RC, logger logging.Logger) {
 	ok, rollbackReason := rc.watchHealth(rcFields, logger)
 	if ok {
-		err := rc.finishTransfer(rcFields, logger)
-		if err == nil {
-			logger.Infoln("Node transfer complete.")
-			return
+		// We retry finishTransfer() up to 10 times in case of a
+		// fleeting transaction building error or network error
+		var err error
+		for i := 0; i < 10; i++ {
+			err = rc.finishTransfer(rcFields, logger)
+			if err == nil {
+				logger.Infoln("Node transfer complete.")
+				return
+			}
+
+			rc.logger.WithError(err).Errorln("could not do final node transfer transaction, retrying")
+
+			backoff := time.Duration(math.Pow(2, float64(i))) * time.Second
```

Comment:
the maximum value of this backoff is something 1024 seconds or ~17 minutes. WDYT about capping it at something like 60s?

Reply:
I think its 512 seconds but to your point I don't really have an opinion. I was unsure how to decide things like the number of allowed retries or the duration of the backoff so I picked them somewhat arbitrarily. 

I am definitely open to changing either. Whats the benefit to a smaller cap like 60s?

Reply:
Fair enough, I had an off by one in my calculations.

For me, it's an argument of marginal utility. What's the likelihood that waiting 8 minutes generates fewer errors than waiting 1 minute? Given that someone is waiting for the node transfer to complete, raising the alarm earlier is better.

Reply:
SGTM. Thoughts on the 10 retries then? I think a similar argument can be made. Whats the likelihood that 10 retries will succeed if ~6 do not?

Reply:
I think 10 is a fine number to start with. We'll retry automatically and be alerted within 10 minutes. I'd like the system to retry on its own until an unacceptable time threshold is met.